tests/lib/workspace.py

```diff
@@ -221,6 +222,25 @@ def destroy(self):
 
         if config._REMOVE_WORKSPACE:
             logger.info(f"Removing workspace {self.working_dir} from disk")
+            # NOTE(jhesketh): go clones repos as read-only. We need to chmod
+            #                 all the files back to writable (in particular,
+            #                 the directories) so that we can remove them
+            #                 without failures or warnings.
+            for root, dirs, files in os.walk(self.working_dir):
```

Comment:
Given that this is `rook from source` specific, could we do that somehow in the rook code?

Reply:
Depends if we want to make rook responsible for removing any files it creates. At the moment I think it's fair to assume that the workspace will be destroyed and not to worry about it at every level.

Additionally I'm moving the go installation into the workspace so skuba or other components can use it (so there may be more to clean up).

I'm inclined to leave it and treat the destroy of workspace as a forced remove.